File: typescript/examples/cancel_order.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * Cancel Order Example
 *
 * Place an order and then cancel it by OID.
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // Place a resting order 3% below mid
  const mid = await sdk.getMid("BTC");
  const limitPrice = Math.floor(mid * 0.97);
  const order = await sdk.buy("BTC", { notional: 11, price: limitPrice, tif: "gtc" });
  console.log(`Placed order OID: ${order.oid}`);

  // Cancel using the order object
  await order.cancel();
  console.log("Cancelled via order.cancel()");

  // Alternative: cancel by OID directly
  // await sdk.cancel(12345, "BTC");
}

main().catch(console.error);

File: python/examples/cancel_order.py
#!/usr/bin/env python3
"""
Cancel Order Example

Place an order and then cancel it by OID.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python cancel_order.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid Cancel Order Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Address: {sdk.address}")

    # Get current price
    mid = sdk.get_mid("BTC")
    print(f"BTC mid price: ${mid:,.2f}")

    # Place a resting order 3% below mid
    limit_price = int(mid * 0.97)
    order = sdk.buy("BTC", notional=11, price=limit_price, tif="gtc")
    print(f"Placed order OID: {order.oid}")

    # Cancel using the order object
    order.cancel()
    print("Cancelled via order.cancel()")

    # Alternative: cancel by OID directly
    # sdk.cancel(12345, "BTC")


if __name__ == "__main__":
    main()
